### PR TITLE
TILES-4788 add plugin registry

### PR DESCRIPTION
**Pull request by Kuznecoff**
Add plugin registry to mapgl plugins namespace

### CODE REVIEW

**Review by ndim3d — APPROVED** [review]
lgtm

File: src/index.ts [hunk 1]
```diff
@@ -1,4 +1,18 @@
 import { Deck2gisLayer } from './deckgl2gisLayer';
 import { initDeck2gisProps } from './utils';
 
+if (typeof window !== 'undefined') {
+    if ('mapgl' in window) {
+        (mapgl as any).Deck2gisLayer = Deck2gisLayer;
+    } else {
+        // Если так вышло, что плагин инициализирован раньше mapgl, поместим его во временную переменную
+        // Из нее уже сам mapgl все положит в себя.
+        if (!(window as any).__mapglPlugins) {
```
Comment (myshov) [inline src/index.ts:10]: I think we have to think about this a little bit more. Now we have a situation when the main project is decoupled from knowledge about existence of new plugin registry. What if someone need to create a new plugin and one doesn't check this plugin. There is a possibility that we end up with one more plugin registry under different name :) I think that the initialization of plugin registry should be placed inside the main project itself.

Comment (myshov) [reply src/index.ts:10]: Maybe even not in a global scope but on the namespace of the map

Comment (myshov) [reply src/index.ts:10]: Discussed in the private messages, it is all correct :ok_hand:

**Review by myshov — APPROVED** [review]

**Pull request merged**